File: Source/FakeItEasy.Examples/RaisingEvents.cs
namespace FakeItEasy.Examples
{
    using FakeItEasy.Examples.ExampleObjects;

    public class RaisingEvents
    {
        public void Raising_event_specifying_both_sender_and_event_arguments()
        {
            var widget = A.Fake<IWidget>();
            widget.WidgetBroke += Raise.With(widget, new WidgetEventArgs("widget name"));
        }

        public void Raising_event_specifying_event_arguments_only()
        {
            var widget = A.Fake<IWidget>();

            // When raising like this the fake object is set as sender.
            widget.WidgetBroke += Raise.With(new WidgetEventArgs("widget name"));
        }
    }
}

File: Source/FakeItEasy.Examples.VB/RaisingEvents.vb
Imports FakeItEasy.Examples.ExampleObjects

Public Class RaisingEvents
    Public Sub Raising_an_event()
        Dim widget = A.Fake(Of IWidget)()

        ' Raise EventHandler(Of T) with the shorter syntax and omitting Go
        AddHandler widget.WidgetBroke, Raise.With(New WidgetEventArgs("foo"))

        ' Raise EventHandler with the shorter syntax and omitting Go
        AddHandler widget.WidgetRunning, Raise.With(New EventArgs())
    End Sub
End Class